### DISCUSSION TITLE
How to suppress verbose output of set-PnPListItem

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/powershell#2763] [Q&A]
I have a long running 🏃 script that does lots of "set-PnPListItem"
Each of them shows something like this:
`FieldValues                        : {[ContentTypeId, 0x0101003DAD9A523623984EBD732073C99173D200CBF30B2F8A6D0C4A8657E9D47E3CE0BA],
                                      [_ModerationComments, ], [FileLeafRef, Sample22.xlsx], [Modified_x0020_By, i:0i.t|00000003-0
                                     000-0ff1-ce00-000000000000|app@sharepoint]…}
AttachmentFiles                    : 
CommentsDisabled                   : 
CommentsDisabledScope              : 
ComplianceInfo                     : 
ContentType                        : Microsoft.SharePoint.Client.ContentType
DisplayName                        : 
GetDlpPolicyTip                    : Microsoft.SharePoint.Client.DlpPolicyTip
EffectiveBasePermissions           : 
EffectiveBasePermissionsForUI      : 
FieldValuesAsHtml                  : Microsoft.SharePoint.Client.FieldStringValues
FieldValuesAsText                  : Microsoft.SharePoint.Client.FieldStringValues
FieldValuesForEdit                 : Microsoft.SharePoint.Client.FieldStringValues
File                               : Microsoft.SharePoint.Client.File
ParentList                         : Microsoft.SharePoint.Client.List
Properties                         : Microsoft.SharePoint.Client.PropertyValues
ServerRedirectedEmbedUri           : https://ecgmcdev.sharepoint.com/teams/2889.001/_layouts/15/Doc.aspx?sourcedoc={04b63a61-9bea- 
                                     4195-9d92-5ff1afaf77b3}&action=interactivepreview
ServerRedirectedEmbedUrl           : https://ecgmcdev.sharepoint.com/teams/2889.001/_layouts/15/Doc.aspx?sourcedoc={04b63a61-9bea- 
                                     4195-9d92-5ff1afaf77b3}&action=interactivepreview
Client_Title                       :
Versions                           :
FirstUniqueAncestorSecurableObject : Microsoft.SharePoint.Client.SecurableObject
HasUniqueRoleAssignments           :
RoleAssignments                    :
Context                            : PnP.Framework.PnPClientContext
Tag                                :
Path                               : Microsoft.SharePoint.Client.ObjectPathIdentity
ObjectVersion                      : 7
ServerObjectIsNull                 : False
TypedObject                        : Microsoft.SharePoint.Client.ListItem`

How can I stop it from showing?
Thank you

**Comment by AndersRask** [comment]
Use
`| Out-null ` 

after the set-

**Reply by Ofer-Gal** [reply]
Thanks. that looks like a solution.

**Reply by AndersRask** [reply]
to be clear, what you are seeing is not verbose output.
You are seeing the output stream from the script. Assigning a value to the cmdlet or piping to Out-Null if you dont need it, is usually considered a best practice, especially because such input can really mess up stuff like Runbooks in Automation accounts.

Verbose output is usually not shown, unless you enable it using the `$VerbosePreference  = "Continue"`

You can find all preference variables like this
```
Get-Variable | ? name -like *preference

Name                           Value
----                           -----
ConfirmPreference              High
DebugPreference                SilentlyContinue
ErrorActionPreference          Continue
InformationPreference          SilentlyContinue
ProgressPreference             Continue
VerbosePreference              SilentlyContinue
WarningPreference              Continue
WhatIfPreference               False
```

To read more on the different output streams in PowerShell check this link 
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_output_streams?view=powershell-7.3

**Reply by Ofer-Gal** [reply]
I see VerbosePreference              SilentlyContinue too and | Out-null  is workin fine
Thank you
